Packet++/src/IPv6Extensions.cpp

```diff
@@ -263,15 +263,15 @@ IPv6AuthenticationHeader::IPv6AuthenticationHeader(uint32_t securityParametersIn
 
 	if (integrityCheckValueLen > 0 && integrityCheckValue != NULL)
 	{
-		uint8_t* icvPtr = (uint8_t*)(getDataPtr() + sizeof(ipv6_authentication_header));
+		uint8_t* icvPtr = getDataPtr() + sizeof(ipv6_authentication_header);
```

Comment:
why did you remove the casting?

Reply:
When a constant is added to a pointer the expression result will have the same type as pointer has.